src/lib/math/bigint/bigint.h

```diff
@@ -567,144 +574,156 @@ class BOTAN_PUBLIC_API(2, 0) BigInt final {
+       */
       void grow_to(size_t n) const { m_data.grow_to(n); }
 
       void resize(size_t s) { m_data.resize(s); }
 
       /**
-     * Fill BigInt with a random number with size of bitsize
-     *
-     * If \p set_high_bit is true, the highest bit will be set, which causes
-     * the entropy to be \a bits-1. Otherwise the highest bit is randomly chosen
-     * by the rng, causing the entropy to be \a bits.
-     *
-     * @param rng the random number generator to use
-     * @param bitsize number of bits the created random value should have
-     * @param set_high_bit if true, the highest bit is always set
-     */
+       * Fill BigInt with a random number with size of bitsize
+       *
+       * If \p set_high_bit is true, the highest bit will be set, which causes
+       * the entropy to be \a bits-1. Otherwise the highest bit is randomly chosen
+       * by the rng, causing the entropy to be \a bits.
+       *
+       * @param rng the random number generator to use
+       * @param bitsize number of bits the created random value should have
+       * @param set_high_bit if true, the highest bit is always set
+       */
       void randomize(RandomNumberGenerator& rng, size_t bitsize, bool set_high_bit = true);
 
       /**
-     * Store BigInt-value in a given byte array
-     * @param buf destination byte array for the integer value
-     */
-      void binary_encode(uint8_t buf[]) const;
+       * Store BigInt-value in a given byte array
+       * @param buf destination byte array for the integer value
+       */
+      void binary_encode(uint8_t buf[]) const {
+         // assumes that `buf` points to a memory region with enough bytes
+         binary_encode(std::span{buf, bytes()});
+      }
 
       /**
-     * Store BigInt-value in a given byte array. If len is less than
-     * the size of the value, then it will be truncated. If len is
-     * greater than the size of the value, it will be zero-padded.
-     * If len exactly equals this->bytes(), this function behaves identically
-     * to binary_encode.
-     *
-     * @param buf destination byte array for the integer value
-     * @param len how many bytes to write
-     */
-      void binary_encode(uint8_t buf[], size_t len) const;
+       * Store BigInt-value in a given byte array
+       * @param buf destination byte array for the integer value
+       * @param len how many bytes to write
+       */
+      void binary_encode(uint8_t buf[], size_t len) const { binary_encode(std::span{buf, len}); }
 
       /**
-     * Read integer value from a byte array with given size
-     * @param buf byte array buffer containing the integer
-     * @param length size of buf
-     */
-      void binary_decode(const uint8_t buf[], size_t length);
+       * Store BigInt-value in a given byte array. If the buffer length
+       * is less than the size of the value, then it will be truncated.
+       * If it is greater than the size of the value, it will be zero-padded.
```

Comment:
@randombit The implementation didn't seem to do actually ensure a zero-padding. I would have expected a call to `clear_mem()` at least. I think that needs to be addressed somehow, but I'm not sure about the implications.

Similarly, in `BigInt::encode_words()`.

Reply:
I’ll look into what’s going with this